compiler/runtime/TRRelocationRuntime.hpp

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
```

Comment:
I think this file should be named `RelocationRuntime.hpp`  for it to fit in with the extensible classes model.

Reply:
This is named TRRelocationRecord to avoid breaking the build of the cmake, because otherwise it would shadow / pick up the wrong file. Thank you for the question!

Reply:
Can you document that is what you're doing here (in the file), please?

Reply:
Thank you Mark, I've added  the comments, will publish them in the next revision.